include/ck/tensor_operation/gpu/device/impl/device_gemm_xdl_cshuffle.hpp

```diff
@@ -397,162 +130,80 @@ struct DeviceGemm_Xdl_CShuffle : public DeviceGemm<ALayout,
         LoopSched,
         PipelineVer>;
 
-    // Argument
-    struct Argument : public BaseArgument
+    struct Argument : public GridwiseGemm::Argument
     {
-        Argument(const ADataType* p_a_grid,
-                 const BDataType* p_b_grid,
-                 CDataType* p_c_grid,
-                 index_t MRaw,
-                 index_t NRaw,
-                 index_t KRaw,
-                 index_t StrideA,
-                 index_t StrideB,
-                 index_t StrideC,
-                 AElementwiseOperation a_element_op,
-                 BElementwiseOperation b_element_op,
-                 CElementwiseOperation c_element_op)
-            : p_a_grid_{p_a_grid},
-              p_b_grid_{p_b_grid},
-              p_c_grid_{p_c_grid},
-              a_grid_desc_ak0_m_ak1_{DeviceOp::MakeAGridDescriptor_AK0_M_AK1(MRaw, KRaw, StrideA)},
-              b_grid_desc_bk0_n_bk1_{DeviceOp::MakeBGridDescriptor_BK0_N_BK1(KRaw, NRaw, StrideB)},
-              c_grid_desc_m_n_{DeviceOp::MakeCGridDescriptor_M_N(MRaw, NRaw, StrideC)},
-              c_grid_desc_mblock_mperblock_nblock_nperblock_{},
-              block_2_ctile_map_{GridwiseGemm::MakeDefaultBlock2CTileMap(c_grid_desc_m_n_)},
-              a_element_op_{a_element_op},
-              b_element_op_{b_element_op},
-              c_element_op_{c_element_op},
-              kraw_{KRaw}
+        using Parent = typename GridwiseGemm::Argument;
+
+        Argument(const ADataType* p_a_grid_,
+                 const BDataType* p_b_grid_,
+                 CDataType* p_c_grid_,
+                 index_t M_,
+                 index_t N_,
+                 index_t K_,
+                 index_t StrideA_,
+                 index_t StrideB_,
+                 index_t StrideC_)
+            : Parent(M_, N_, K_, StrideA_, StrideB_, StrideC_),
+              p_a_grid{p_a_grid_},
+              p_b_grid{p_b_grid_},
+              p_c_grid{p_c_grid_}
```

Comment:
I think the better place for those pointers would be in `GridwiseGemm::Argument` object. This way you won't have to pass them back-and-forth manually through the kernel function.

Reply:
okay, I'll move them back to `GridwiseGemm::Argument`

Reply:
I had splitted the giant type `GridwiseGemm::Argument` into `GridwiseGemm::Argument` (smaller) and `GridwiseGemm::Problem` types, and allow device operators to choose one of them to derive from.

```c++
struct Problem {
   index_t M;
   index_t N;
   index_t K;
};

struct Argument : public BaseArgument, public Problem {
   const FloatAB* p_a_grid;
   const FloatAB* p_b_grid;
   FloatC* p_c_grid;
};
```

In the case of `DeviceCGemm_4Gemm_Xdl_CShuffle<>`, we only need to inherit from `Problem` and add pointer data members